internal/server/auth/method/http.go

```diff
@@ -90,7 +92,7 @@ func (m Middleware) ForwardResponseOption(ctx context.Context, w http.ResponseWr
 // The payload is then also encoded as a http cookie which is bound to the callback path.
 func (m Middleware) Handler(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		provider, method, match := parts(r.URL.Path)
+		prefix, provider, method, match := parts(r.URL.Path)
```

Comment:
Take it or leave it:
I reflected a bit on this, after you raised the very valid point that there there is the subtle difference of a provider for OIDC and not for Github and figured, which makes this cutting and prefixing fiddly; I think there might be just a clearer way to express all this.

In hindsight, all this path matching is only really attempting to do two things:
1. Match valid `authorize` request paths (one of OIDC and one for Google).
2. Replace the `authorize` path entry with the word `callback` for the callback path.

I previously had overengineered this with pulling out the provider, but I cant see where that is getting used.

I wonder if the inconsistencies and fiddly reconstruction could all be simplified with just a bit of inline code that does just this:

e.g. 

```go
func (m Middleware) Handler(next http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		prefix, method := path.Split(r.URL.Path)
		if !((strings.HasPrefix(prefix, oidcPrefix) || strings.HasPrefix(prefix, ghPrefix)) && method == "authorize") {
			next.ServeHTTP(w, r)
			return
		}
		
		// ...
		
		
			// bind state cookie to callback handler
			Path:     path.Join(prefix, "callbak"),
```

Reply:
@GeorgeMac Awesome suggestion!